sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/SpanLimits.scala

```diff
@@ -0,0 +1,204 @@
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/trace/sdk/#span-limits]]
+  */
+sealed trait SpanLimits {
+
+  /** The max number of attributes per span. */
+  def maxNumberOfAttributes: Int
+
+  /** The max number of events per span. */
+  def maxNumberOfEvents: Int
+
+  /** The max number of links per span. */
+  def maxNumberOfLinks: Int
+
+  /** The max number of attributes per event. */
+  def maxNumberOfAttributesPerEvent: Int
+
+  /** The max number of attributes per link. */
+  def maxNumberOfAttributesPerLink: Int
+
+  /** The max number of characters for string attribute values. For string array
+    * attribute values, applies to each entry individually.
+    */
+  def maxAttributeValueLength: Int
+
+  override final def hashCode(): Int =
+    Hash[SpanLimits].hash(this)
+
+  override final def equals(obj: Any): Boolean =
+    obj match {
+      case other: SpanLimits => Hash[SpanLimits].eqv(this, other)
+      case _                 => false
+    }
+
+  override final def toString: String =
+    Show[SpanLimits].show(this)
+}
+
+object SpanLimits {
+
+  private object Defaults {
+    val MaxNumberOfAttributes = 128
+    val MaxNumberOfEvents = 128
+    val MaxNumberOfLinks = 128
+    val MaxNumberOfAttributesPerEvent = 128
+    val MaxNumberOfAttributesPerLink = 128
+    val MaxAttributeValueLength = Int.MaxValue
+  }
+
+  /** Builder for [[SpanLimits]] */
+  sealed trait Builder {
+
+    /** Sets the max number of attributes per span. */
+    def withMaxNumberOfAttributes(value: Int): Builder
+
+    /** Sets the max number of events per span. */
+    def withMaxNumberOfEvents(value: Int): Builder
+
+    /** Sets the max number of links per span. */
+    def withMaxNumberOfLinks(value: Int): Builder
+
+    /** Sets the max number of attributes per event. */
+    def withMaxNumberOfAttributesPerEvent(value: Int): Builder
+
+    /** Sets the max number of attributes per link. */
+    def withMaxNumberOfAttributesPerLink(value: Int): Builder
+
+    /** Sets the max number of characters for string attribute values. For
+      * string array attribute values, applies to each entry individually.
+      */
+    def withMaxAttributeValueLength(value: Int): Builder
+
+    /** Creates a [[SpanLimits]] with the configuration of this builder. */
+    def build: SpanLimits
+  }
+
+  private val Default: SpanLimits =
+    builder.build
+
+  /** Creates a [[Builder]] for [[SpanLimits]] using the default limits.
+    */
+  def builder: Builder =
+    BuilderImpl(
+      maxNumberOfAttributes = Defaults.MaxNumberOfAttributes,
+      maxNumberOfEvents = Defaults.MaxNumberOfEvents,
+      maxNumberOfLinks = Defaults.MaxNumberOfLinks,
+      maxNumberOfAttributesPerEvent = Defaults.MaxNumberOfAttributesPerEvent,
+      maxNumberOfAttributesPerLink = Defaults.MaxNumberOfAttributesPerLink,
+      maxAttributeValueLength = Defaults.MaxAttributeValueLength
+    )
+
+  /** Creates a [[SpanLimits]] using the given limits. */
+  def create(
```

Comment:
Currently, for pure constructors we use `apply`:
```suggestion
  def apply(
```

Reply:
Got it, renamed.